net/blockwise/blockwise.go

```diff
@@ -664,7 +664,128 @@ func (b *BlockWise) getSendedRequest(token message.Token) Message {
 	return nil
 }
 
-func (b *BlockWise) processReceivedMessage(w ResponseWriter, r Message, maxSzx SZX, next func(w ResponseWriter, r Message), blockType message.OptionID, sizeType message.OptionID) error {
+func (b *BlockWise) handleObserveResponse(r, sendedRequest Message, expire time.Time) (message.Token, time.Time, error) {
+	if !isObserveResponse(r) {
+		return r.Token(), expire, nil
+	}
+
+	// https://tools.ietf.org/html/rfc7959#section-2.6 - performs GET with new token.
+	if sendedRequest == nil {
+		return nil, expire, fmt.Errorf("observation is not registered")
+	}
+	token, err := message.GetToken()
+	if err != nil {
+		return nil, expire, fmt.Errorf("cannot get token for create GET request: %w", err)
+	}
+	expire = time.Now().Add(b.expiration) // context of observation can be expired.
+	bwSendedRequest := b.newSendRequestMessage(sendedRequest, true)
+	bwSendedRequest.SetToken(token)
+	err = b.bwSendedRequest.store(bwSendedRequest)
+	if err != nil {
+		return nil, expire, fmt.Errorf("cannot store sended request %v: %v", bwSendedRequest.String(), err)
+	}
+
+	return token, expire, nil
+}
+
+func (b *BlockWise) messageFromReceivedCache(w ResponseWriter, r Message, szx, maxSzx SZX, next func(w ResponseWriter, r Message),
+	more bool, tokenStr uint64, expire time.Time) (Message, SZX, error) {
+	var cachedReceivedMessageGuard interface{}
+	e := b.receivingMessagesCache.Load(tokenStr)
+	if e != nil {
+		cachedReceivedMessageGuard = e.Data()
+	}
+
+	var msgGuard *messageGuard
+	if cachedReceivedMessageGuard != nil {
+		msgGuard = cachedReceivedMessageGuard.(*messageGuard)
+		err := msgGuard.Acquire(msgGuard.Context(), 1)
+		if err != nil {
+			return nil, 0, fmt.Errorf("handlerReceivedCache: cannot lock message: %v", err)
```

Comment:
the label in `fmt.Errorf` `handlerReceivedCache` doesn't correspond to the function name

Reply:
done.

nits: `*sended*` -> `*sent*` to be changed in logs, error messages, function names and var names across this library.

Reply:
Yeah, I've run `misspell` over the code, but there's still plenty of spelling mistakes left. Perhaps there's some other tool I can integrate to check the code. Anyway, feel free to fix a mistake when you see it.